### PR TITLE
Move to {box}

### PR DESCRIPTION
**Pull request by caldwellst**
Zack, here's a very large PR which moves us to a modern, modular code base.

This addresses #24 by creating standalone modules that all run on their own and have full documentation. I have completely documented every module folder and file I think quite well, of course note when there are missing parts, but I hope this makes the entire setup very clear.

I have addressed #25 by putting into these new modules a system that ensures you cannot save out data, send emails, or access the OpenAI API unless you are very intentional. This means you can safely develop without fear!

We have updated to the new packages for #22, but I have not moved to {capsule}. Looking into it, {capsule} is a very old school and weird system where they require all `library()` calls in a single file. I don't think this is good practice and preferred going with the {box} system. We can still look into getting rid of {renv} but I haven't yet as I am not sure how best to get rid of without losing the caching functionality.

I found this very exciting, so hope you like it!

### CODE REVIEW

**Comment by zackarno** [comment]
Really awesome stuff! I like `{box}` so far and the `gmas_test_run()` really makes troubleshooting much easier. Great improvements all around.

There are a couple potential issues I highlighted in the review. The main issue seems now to be with `{ripc}` (unrelated to this repo - I posted here https://github.com/OCHA-DAP/ripc/issues/6#issue-1901834581

Once these little issues are addressed i think it's good to merge. While the `{ripc}` issue is somewhat separate,  I think it's disrupting the processing here. Therefore, I think the order/sequence of work/merging is best left to you!


Additionally, if we can ever figure out a way to get rid of these types of errors it would be great for troubleshooting. We talked briefly about these on slack so I don't think it's a priority for this PR.

```
Error in box::use(gs = ../../utils/google_sheets) : 
  unable to load module “../../utils/google_sheets”; not found in “/{path}/pa-global-monitoring”
```

**Comment by caldwellst** [comment]
Hmm maybe you didn't submit the comments/code review? Not seeing it yet on this PR.

**Comment by zackarno** [comment]
> Hmm maybe you didn't submit the comments/code review? Not seeing it yet on this PR.

ah yes, most have forgot click that - anyways just did

File: src/indicators/ipc/update_ipc.R [hunk 1]
```diff
@@ -1,23 +1,18 @@
...
-    "src",
-    "utils",
-    "get_country_names.R"
-  )
-)
+# internal utilities
+# first set the root search path for utilities
+box::use(gs = ../../utils/google_sheets)
+box::use(../../utils/ai_summarizer[ai_summarizer])
+box::use(../../utils/get_country_names[get_country_names])
```
[the remark this answers is not in the document]
Comment (caldwellst) [reply src/indicators/ipc/update_ipc.R:15]: Doh! Good catch.

File: src/indicators/ipc/update_ipc.R [hunk 2]
```diff
@@ -33,109 +28,110 @@ source(
...
-  mutate(
+  ) |>
+  dplyr$mutate(
     iso3 = ifelse(
       country == "LAC",
       country,
-      countrycode(country, origin = "iso2c", destination = "iso3c")
+      countrycode$countrycode(country, origin = "iso2c", destination = "iso3c")
     )
-  ) %>%
```
[the remark this answers is not in the document]
Comment (caldwellst) [reply src/indicators/ipc/update_ipc.R:46]: Yeah, good idea to remove explicitly.

File: src/utils/google_sheets.R [hunk 3]
```diff
@@ -1,48 +1,109 @@
...
+#' Searches `all_files` to find the row that matches `name`. These are passed
+#' directly to read from and update the Google Sheets via API.
+#'
+#' @param name Name of the Google Sheet.
+#'
+#' @returns Google Drive row that matches the name, which can be used to download
+#'     to download and update the file.
 get_gs_file <- function(name) {
-  dplyr::filter(all_files, stringr::str_starts(name, !!name))
+  dplyr$filter(all_files, stringr::str_starts(name, !!name))
```
[the remark this answers is not in the document]
Comment (caldwellst) [reply src/utils/google_sheets.R:96]: lol yes damnit, good catch. I've searched everywhere and removed them all now.

**Pull request merged**